packages/nocms-plugin-file-resources/test/index.spec.js

```diff
@@ -0,0 +1,136 @@
+import fs from 'fs';
+import { promisify } from 'util';
+import { expect } from 'chai';
+import { activate } from '../src/index.js';
+
+const readFileAsync = promisify(fs.readFile);
+
+let resourceProviders = [];
+
+let fakeFiles = [];
+let writtenFiles = [];
+
+let filePaths = [
+	'test/test-files/fakeJSFileName.js',
+	'test/test-files/fakeHTMLFileName.html',
+	'test/test-files/fakeExtFileName.fakeExtension'
+];
+
+async function fileToBuffer(path) {
+	return readFileAsync(path);
+}
+
+function readFileFake(fakePath) {
+	return new Promise((resolve, reject) => {
+		const fakeFile = fakeFiles.find(fakeFile => fakeFile.path === fakePath);
+		if (!fakeFile) {
+			reject('readFileFakeError');
+		}
+		resolve(fakeFile.data);
+	});
+}
+
+function writeFileFake(filePath, data, options) {
+	writtenFiles[filePath] = data;
+	return Promise.resolve();
+}
+
+function findFilesFake(globPattern) {
+	return new Promise(resolve => {
+		resolve(fakeFiles.map(fakeFile => fakeFile.path));
+	});
+}
+
+describe('nocms-plugin-file-resources', async () => {
+	let pluginActivationContext;
+	before(async () => {
+		await Promise.all(filePaths.map(async (filePath) => {
```

Comment:
I think you should declare the variable of the result and push that to fakeFiles afterwards. 

It is odd behaviour to `.map` as a iterator; https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map